tests/test_nbdserver.c

```diff
@@ -0,0 +1,82 @@
+#ifndef _LARGEFILE64_SOURCE
+#define _LARGEFILE64_SOURCE
+#endif /*_LARGEFILE64_SOURCE */
+#include <sys/types.h>
+
+#include "unity.h"
+
+#include <string.h>
+#include <assert.h>
+#include <uuid/uuid.h>
+#include <stdlib.h>
+
+#include "drivers/tapdisk-nbdserver.h"
+
+#include "mock_tapdisk-log.h"
+#include "mock_tapdisk-server.h"
+#include "mock_tapdisk-utils.h"
+#include "mock_tapdisk-vbd.h"
+#include "mock_tapdisk-fdreceiver.h"
+
+unsigned PAGE_SIZE = 1 << 12;
+
+td_nbdserver_t server;
+td_nbdserver_client_t *client;
+
+void setUp(void) {
+
+	tlog_syslog_Ignore();
+	tapdisk_server_unregister_event_Ignore();
+
+	memset(&server, 0, sizeof server);
+	INIT_LIST_HEAD(&server.clients);
+
+	client = tapdisk_nbdserver_alloc_client(&server);
+	assert(client);
+}
+
+void tearDown(void) {
```

Comment:
So, how do know in tapdisk when a client structure is pointing to a valid location or not?
If tapdisk_nbdserver_free_client just frees the memory, how do we know if the pointer is
valid or not?